core/domain/suggestion_services_test.py

```diff
@@ -2084,6 +2094,1271 @@ def test_create_translation_contribution_stats_from_model(self) -> None:
+            )
+        )
+        # Assert translation review stats after the review.
+        # At this point we can confirm that there should be an associated
+        # translation review stat object for the given IDs since we have
+        # called update_translation_review_stats function to create/update
+        # translation review stats.
+        assert translation_review_stats_model is not None
+        self.assertEqual(
+            translation_review_stats_model.accepted_translations_count,
+            2
+        )
+        self.assertEqual(
+            (
+                translation_review_stats_model
+                .reviewed_translation_word_count
+            ),
+            4
+        )
+        self.assertEqual(
+            translation_review_stats_model
+            .accepted_translations_with_reviewer_edits_count,
+            1
+        )
+        # Assert translation contribution stats after the review.
+        # At this point we can confirm that there should be an associated
+        # translation contribution stat object for the given IDs since we have
+        # called update_translation_contribution_stats_at_submission function
+        # to create/update translation contribution stats.
+        assert translation_contribution_stats_model is not None
+        self.assertEqual(
+            translation_contribution_stats_model.submitted_translations_count,
+            2
+        )
+        self.assertEqual(
+            (
+                translation_contribution_stats_model
+                .submitted_translation_word_count
+            ),
+            6
+        )
+        self.assertEqual(
+            translation_contribution_stats_model.accepted_translations_count,
+            2
+        )
+        self.assertEqual(
+            (
+                translation_contribution_stats_model
+                .accepted_translations_without_reviewer_edits_count
+            ),
+            1
+        )
+
+    def test_update_question_contribution_stats_when_submitting(self) -> None:
+        # Steps required in the setup phase before testing.
+        # 1. Save new skills.
+        # 2. Save a topic assigning skills for it.
+        # 4. Create a question suggestion.
+        skill_id_1 = skill_services.get_new_skill_id()
+        skill_id_2 = skill_services.get_new_skill_id()
+        self.save_new_skill(
+            skill_id_1, self.author_id, description='description')
+        self.save_new_skill(
+            skill_id_2, self.author_id, description='description')
+        topic_id = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            topic_id, 'topic_admin', name='Topic1',
+            abbreviated_name='topic-three', url_fragment='topic-three',
+            description='Description',
+            canonical_story_ids=[],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[skill_id_1, skill_id_2],
+            subtopics=[], next_subtopic_id=1)
```

Comment:
This also seems duplicated across the question suggestion tests, so we should extract it into a helper function. In general, try to extract out the uninteresting parts of tests, e.g. prerequisite setup that's the reader doesn't really care about in the test case.

Reply:
Done